linode/go-metadata
# fix: Fully disable root login for E2E testing instances

## 📝 Description

This change updates the E2E test hardening script to fully disable root login on test instances. This is necessary because `disable_root` does not seem to enforce the `PermitRootLogin no` SSH daemon config option.

## ✔️ How to Test

1. Run the E2E test target to provision an E2E testing Linode
- `make e2e`
2. Once the E2E suite has run, attempt to SSH into the test instance as root. (NOTE: The IP of the test instance can be grabbed from Cloud Manager)
- `ssh root@MYINSTANCEIP`
3. Observe the authentication immediately fails
4. Attempt to login using the `linodedx` account.
- `ssh linodedx@MYINSTANCEIP`
5. Observe the login passes as expected.
